gcc/rust/checks/errors/rust-const-checker.h

```diff
@@ -71,6 +71,9 @@ class ConstChecker : public HIRFullVisitor
   void check_default_const_generics (
     std::vector<std::unique_ptr<GenericParam>> &param, ConstGenericCtx context);
 
+  /* Are we currently in a const context? */
+  bool in_const_context = false;
```

Comment:
This is not necessary and also bug prone. You can read the whole discussion [here](https://github.com/Rust-GCC/gccrs/pull/1410#discussion_r930960483) which is what led us to come up with the `StackedContext` class, which you can find right under your boolean :) You can check whether or not you are in a constant context (like a constant function) by calling `const_context.is_in_context ()`, which will return a boolean.

The class is defined [here](https://github.com/Rust-GCC/gccrs/blob/master/gcc/rust/util/rust-stacked-contexts.h), in `gcc/rust/util/rust-stacked-context.h`